engine/runtime/src/main/java/org/enso/interpreter/node/callable/IndirectInvokeCallableNode.java

```diff
@@ -52,6 +57,37 @@ public abstract Object execute(
       InvokeCallableNode.ArgumentsExecutionMode argumentsExecutionMode,
       BaseNode.TailStatus isTail);
 
+  @Specialization(guards = "warnings.hasWarnings(warning)")
+  Object invokeWithWarnings(
+      Object warning,
```

Comment:
My first thought here was that the specialization guard is not disjunctive - how do we ensure for example that `invokeFunction` will not still get a value containing warnings if it was selected before.

But I realised that values with warnings will never have a type `Function` directly, they will have a type `WithWarnings`, so they can only match this branch.

So it looks ok.

But because this is a non-trivial implicit assumption, I'd suggest to add a short comment note around here.

Maybe
```suggestion
  /** Only this specialization will match objects with warnings attached as they will have type `WithWarnings` which is not a subtype of `Function` or other types from specializations below; and the last specialization is a `Fallback`. */
  @Specialization(guards = "warnings.hasWarnings(warning)")
  Object invokeWithWarnings(
      Object warning,
```